### ISSUE TITLE
Type.Record(Type.TemplateLiteral(...), ...) is wrong on both typing and schema

### ISSUE THREAD

**Issue by m-ronchi** [sinclairzx81/typebox#380]
```
const Foo = Type.Record(Type.TemplateLiteral([Type.String(), Type.Literal("_foo"),]), Type.String());
type Foo = Static<typeof Foo>;
console.log(Foo);
```
produces this schema: 
```
{
  type: 'object',
  properties: { '.*_foo': { type: 'string', [Symbol(TypeBox.Kind)]: 'String' } },
  required: [ '.*_foo' ],
  [Symbol(TypeBox.Kind)]: 'Object'
}
```
and this type:
![image](https://user-images.githubusercontent.com/35332423/231774616-9702c30f-2f1c-4b44-93c3-8d345db1f165.png)

while I would expect this schema:
```
{
  type: 'object',
  patternProperties: { '^.*_foo$': { type: 'string', [Symbol(TypeBox.Kind)]: 'String' } },
  [Symbol(TypeBox.Kind)]: 'Object'
}
```
(`patternProperties` instead of `properties`, also pattern needs to be anchored)
and this type:
```
type Foo = {
  [x: `${string}_foo`]: string;
}
```

note: `Type.TemplateLiteral` alone works

**Comment by sinclairzx81** [comment]
@m-ronchi Thanks for reporting, will look into this tomorrow

Cheers!
S

**Comment by sinclairzx81** [comment]
@m-ronchi This should be resolved on `0.27.5`. The related PR can be found https://github.com/sinclairzx81/typebox/pull/381

Will leave this issue open for a while to give you a chance to review. Quite keen to have users report issues early for this new functionality. Template literals are a very new and a fairly sophisticated piece of functionality. Catching all the edge cases will be good (particularly for literals with infinite expansions (like TString)). So if you spot anything else out of order, please don't hesitate to let me know :)

Cheers
S

**Comment by sinclairzx81** [comment]
@m-ronchi Hey, will close this off as this issue should be resolved on #381 (inclusive of the correct template literal record key). But if you run into any issues, please don't hesitate to let me know :)

Cheers!
S
